app/scripts/modules/core/healthCounts/HealthCounts.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+import autoBindMethods from 'class-autobind-decorator';
+import { isEqual } from 'lodash';
+
+import { InstanceCounts } from 'core/domain';
+import { Tooltip } from 'core/presentation/Tooltip';
+
+import './healthCounts.less';
+
+interface IProps {
+  container: InstanceCounts;
+  additionalLegendText?: string;
+  legendPlacement?: 'top' | 'bottom' | 'left' | 'right';
+};
+
+interface IState {
+  percentLabel: string;
+  statusClass: string;
+  total: number;
+};
+
+@autoBindMethods
+export class HealthCounts extends React.Component<IProps, IState> {
+  public static defaultProps: Partial<IProps> = {
+    legendPlacement: 'top',
+    container: {} as InstanceCounts
+  };
+
+  constructor(props: IProps) {
+    super(props);
+    this.state = this.calculatePercent(props.container);
```

Comment:
Is it worth coming up with a standard naming convention for returning state from props?

I've been doing a similar thing: `this.state = this.getState(props)` and then in `componentWillReceiveProps`: `this.setState(this.getState(props))

Other possibilities: `computeState()`, `buildState()`, `stateFromProps()`